llvm/lib/MC/MCParser/ELFAsmParser.cpp

```diff
@@ -697,6 +697,15 @@ bool ELFAsmParser::ParseSectionArguments(bool IsPush, SMLoc loc) {
       getContext().getELFSection(SectionName, Type, Flags, Size, GroupName,
                                  IsComdat, UniqueID, LinkedToSym);
   getStreamer().switchSection(Section, Subsection);
+
+  // Section alignment of 4 if an AArch64 instruction is used when $x mapping
+  // symbol is added Match GNU Assembler
+  const Triple &TT = getContext().getTargetTriple();
+  if ((Section->getFlags() & ELF::SHF_EXECINSTR) && (TT.isAArch64())) {
+    if (Section->getAlign() < 4)
+      getStreamer().emitValueToAlignment(Align(4));
+  }
```

Comment:
You don't need to do this in target-independent code - `MCStreamer::switchSection` will call `MCStreamer::changeSection` (the latter is `virtual`, and `AArch64ELFStreamer` has an override of it already, where you can add logic like this (Note this code is also handling the mapping symbols):

 https://github.com/llvm/llvm-project/blob/main/llvm/lib/Target/AArch64/MCTargetDesc/AArch64ELFStreamer.cpp#L191-L205

Reply:
I have moved the logic into `MCStreamer::changeSection` of the AArch64ELFStreamer as suggested